File: Code/WebSvc.dotNet/WebSvc.dotNet/Repositories/MongoAdapter.cs
using Model;
using ThirdPartyCode;

namespace Repositories
{
    /// <summary>
    /// Adapter pattern to wrap 3rd party MongoDb
    /// </summary>
    public class MongoAdapter : IDb
    {
        private readonly MongoDb _db = new MongoDb();

        public void UpdateBirth(Person p)
            => _db.externalUpdateBirth(p.Id, p.Age);
    }
}

File: Code/WebSvc.Java/src/main/java/repositories/MongoAdapter.java
package repositories;

import models.Person;
import thirdparty.MongoDb;

/**
 * Adapter pattern to wrap 3rd party MongoDb
 */
public class MongoAdapter implements IDb {
	
	private MongoDb _db = new MongoDb();

	@Override
	public void updateBirth(int key, int age) throws Exception {
		this._db.externalUpdateBirth(key, age);
	}

	@Override
	public void updateBirth(Person p) throws Exception {
		this._db.externalUpdateBirth(p.getId(), p.getAge());
	}

}
